components/ai_chat/core/browser/utils.cc

```diff
@@ -80,6 +84,20 @@ void SetUserOptedIn(PrefService* prefs, bool opted_in) {
   }
 }
 
+void ResetBadModelKeys(PrefService* prefs) {
```

Comment:
Can't this be in `ai_chat::prefs` like the other migration function? In fact, couldn't it be called from `ai_chat::prefs::MigrateProfilePrefs`?

Reply:
shouldn't prefs.h be self-contained? adding this logic inside prefs would require `core/browser/model.h` which would then need to depend on `//ai_chat/core/browser`? i don't see any files in common directory do this.

Reply:
`core/browser` is ok for prefs, `content/browser` is not. Fine to wait until ModelService and do all the prefs and migration in there.